pkg/controller/machine/machine_controller.go

```diff
@@ -116,6 +117,7 @@ type Reconciler struct {
 	satelliteSubscriptionManager     rhsm.SatelliteSubscriptionManager
 
 	useOSM        bool
+	externalCSI   bool
```

Comment:
Why this needed? If you would like to know if the external cluster is used then you can check the annotation on the machine: 
```v1.kubelet-flags.machine-controller.kubermatic.io/ExternalCloudProvider: "true"```

Reply:
I didn't consider the existence of that annotation, I'll use it.